src/main/scala/com/fortysevendeg/android/sqlite/statement/SQLDroidPreparedStatement.scala

```diff
@@ -0,0 +1,289 @@
+package com.fortysevendeg.android.sqlite.statement
+
+import java.io.{InputStream, Reader}
+import java.math.BigDecimal
+import java.net.URL
+import java.sql._
+import java.util.Calendar
+
+import com.fortysevendeg.android.sqlite.{SQLDroidConnection, _}
+import com.fortysevendeg.android.sqlite.logging.{AndroidLogWrapper, LogWrapper}
+import com.fortysevendeg.android.sqlite.util.StreamUtils._
+import StatementInfo._
+
+import scala.util.{Failure, Success, Try}
+
+class SQLDroidPreparedStatement(
+  val sql: String,
+  sqlDroidConnection: SQLDroidConnection,
+  val columnGenerated: Option[String] = None,
+  arguments: PreparedStatementArguments = new PreparedStatementArguments,
+  logWrapper: LogWrapper = new AndroidLogWrapper())
+  extends SQLDroidStatement(sqlDroidConnection, columnGenerated, logWrapper)
+  with PreparedStatement {
+
+  val notInPreparedErrorMessage = "This method cannot be called on a PreparedStatement"
+
+  val maximumSizeErrorMessage = "Byte maximum exceeded"
+
+  val invalidLengthErrorMessage = (s: Long) => s"Invalid length: $s"
+
+  val executingSQLErrorMessage = (s: String) => s"Error executing statement $s"
+
+  override def executeQuery(sql: String): ResultSet =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def executeUpdate(sql: String): Int =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def executeUpdate(sql: String, autoGeneratedKeys: Int): Int =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def executeUpdate(sql: String, columnIndexes: scala.Array[Int]): Int =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def executeUpdate(sql: String, columnNames: scala.Array[String]): Int =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def execute(sql: String): Boolean =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def execute(sql: String, autoGeneratedKeys: Int): Boolean =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def execute(sql: String, columnIndexes: scala.Array[Int]): Boolean =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def execute(sql: String, columnNames: scala.Array[String]): Boolean =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def addBatch(sql: String): Unit =
+    throw new SQLException(notInPreparedErrorMessage)
+
+  override def addBatch(): Unit =
+    arguments.addNewEntry()
+
+  override def clearParameters(): Unit =
+    arguments.clearArguments()
+
+  override def execute(): Boolean =
+    executeWithArgs(sql, Some(arguments))
+
+  override def executeQuery(): ResultSet =
+    executeQueryWithArgs(sql, Some(arguments))
+
+  override def executeUpdate(): Int =
+    executeUpdateWithArgs(sql, Some(arguments))
+
+  override def executeBatch(): scala.Array[Int] = withOpenConnection { db =>
+    val updateArray = arguments map { array =>
+      Try(db.execSQL(sql, array)) match {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/47deg/scala-sqlite-droid/file/821582810/issues/source?bid=2442814&fileBranchId=2752104#l80)